### PR TITLE
wplay-gui initial setup issue #43

### PR DESCRIPTION
**Pull request by msk4862**
wplay-gui initial setup #43  with reactJS + electronJS.
Also added a readme file for setup instructions.

### CODE REVIEW

**Comment by rpotter12** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 19
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5104430&bid=16787695)

File: wplay_gui/wplay/src/App.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import React from 'react';
```
Comment (rpotter12) [inline wplay_gui/wplay/src/App.js]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5104430)

File: wplay_gui/wplay/public/electron.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
+const { app, BrowserWindow } = require('electron')
```
Comment (rpotter12) [inline wplay_gui/wplay/public/electron.js]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5104430)

File: wplay_gui/wplay/public/electron.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  }
+})
+
+app.on('activate', () => {
+  // On macOS it's common to re-create a window in the app when the
+  // dock icon is clicked and there are no other windows open.
+  if (BrowserWindow.getAllWindows().length === 0) {
+    createWindow()
+  }
+})
```
Comment (rpotter12) [inline wplay_gui/wplay/public/electron.js]: Codacy Issue found: [Missing semicolon.](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5104430)

File: wplay_gui/wplay/public/electron.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+// initialization and is ready to create browser windows.
+// Some APIs can only be used after this event occurs.
+app.whenReady().then(createWindow)
+
+// Quit when all windows are closed.
+app.on('window-all-closed', () => {
+  // On macOS it is common for applications and their menu bar
+  // to stay active until the user quits explicitly with Cmd + Q
+  if (process.platform !== 'darwin') {
+    app.quit()
```
Comment (rpotter12) [inline wplay_gui/wplay/public/electron.js]: Codacy Issue found: [Missing semicolon.](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5104430)

File: wplay_gui/wplay/public/electron.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  )
+
+  // Open the DevTools.
+  // win.webContents.openDevTools()
+}
+
+// This method will be called when Electron has finished
+// initialization and is ready to create browser windows.
+// Some APIs can only be used after this event occurs.
+app.whenReady().then(createWindow)
```
Comment (rpotter12) [inline wplay_gui/wplay/public/electron.js]: Codacy Issue found: [Missing semicolon.](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5104430)

File: wplay_gui/wplay/public/electron.js [hunk 6]
```diff
@@ -0,0 +1,44 @@
+const { app, BrowserWindow } = require('electron')
+const path = require("path")
```
Comment (rpotter12) [inline wplay_gui/wplay/public/electron.js]: Codacy Issue found: [Missing semicolon.](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5104430)

File: wplay_gui/wplay/public/electron.js [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+const isDev = require("electron-is-dev")
```
Comment (rpotter12) [inline wplay_gui/wplay/public/electron.js]: Codacy Issue found: [Missing semicolon.](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5104430)

File: wplay_gui/wplay/src/index.js [hunk 8]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import './index.css';
```
Comment (rpotter12) [inline wplay_gui/wplay/src/index.js]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5104430)

File: wplay_gui/wplay/public/electron.js [hunk 9]
```diff
@@ -0,0 +1,44 @@
...
+// This method will be called when Electron has finished
+// initialization and is ready to create browser windows.
+// Some APIs can only be used after this event occurs.
+app.whenReady().then(createWindow)
+
+// Quit when all windows are closed.
+app.on('window-all-closed', () => {
+  // On macOS it is common for applications and their menu bar
+  // to stay active until the user quits explicitly with Cmd + Q
+  if (process.platform !== 'darwin') {
```
Comment (rpotter12) [inline wplay_gui/wplay/public/electron.js]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5104430)

File: wplay_gui/wplay/public/electron.js [hunk 10]
```diff
@@ -0,0 +1,44 @@
...
+    app.quit()
+  }
+})
+
+app.on('activate', () => {
```
Comment (rpotter12) [inline wplay_gui/wplay/public/electron.js]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5104430)

File: wplay_gui/wplay/public/electron.js (hunk 2 above)
Comment (rpotter12) [inline wplay_gui/wplay/public/electron.js]: Codacy Issue found: [Missing semicolon.](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5104430)

File: wplay_gui/wplay/public/electron.js [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+    height: 600,
+    webPreferences: {
+      nodeIntegration: true
+    }
+  })
+
+  // and load the index.html of the app.
+  win.loadURL(
+    isDev ? "http://localhost:3000" : `file://${path.join(__dirname, "../build/index.html")}`
+  )
```
Comment (rpotter12) [inline wplay_gui/wplay/public/electron.js]: Codacy Issue found: [Missing semicolon.](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5104430)

File: wplay_gui/wplay/public/electron.js [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+  if (process.platform !== 'darwin') {
+    app.quit()
+  }
+})
+
+app.on('activate', () => {
+  // On macOS it's common to re-create a window in the app when the
+  // dock icon is clicked and there are no other windows open.
+  if (BrowserWindow.getAllWindows().length === 0) {
+    createWindow()
```
Comment (rpotter12) [inline wplay_gui/wplay/public/electron.js]: Codacy Issue found: [Missing semicolon.](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5104430)

File: wplay_gui/wplay/public/electron.js [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+  // win.webContents.openDevTools()
+}
+
+// This method will be called when Electron has finished
+// initialization and is ready to create browser windows.
+// Some APIs can only be used after this event occurs.
+app.whenReady().then(createWindow)
+
+// Quit when all windows are closed.
+app.on('window-all-closed', () => {
```
Comment (rpotter12) [inline wplay_gui/wplay/public/electron.js]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5104430)

File: wplay_gui/wplay/src/index.js [hunk 14]
```diff
@@ -0,0 +1,11 @@
...
+import App from './App';
```
Comment (rpotter12) [inline wplay_gui/wplay/src/index.js]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5104430)

File: wplay_gui/wplay/src/index.js [hunk 15]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
```
Comment (rpotter12) [inline wplay_gui/wplay/src/index.js]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5104430)

File: wplay_gui/wplay/src/index.js [hunk 16]
```diff
@@ -0,0 +1,11 @@
...
+import './index.css';
+import App from './App';
+
+ReactDOM.render(<App />, document.getElementById('root'));
```
Comment (rpotter12) [inline wplay_gui/wplay/src/index.js]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5104430)

File: wplay_gui/wplay/src/App.js [hunk 17]
```diff
@@ -0,0 +1,14 @@
...
+import './App.css';
```
Comment (rpotter12) [inline wplay_gui/wplay/src/App.js]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5104430)

File: wplay_gui/wplay/src/index.js [hunk 18]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
```
Comment (rpotter12) [inline wplay_gui/wplay/src/index.js]: Codacy Issue found: [Strings must use doublequote.](https://app.codacy.com/manual/rpotter12/whatsapp-play/pullRequest?prid=5104430)

**Comment by priyansh19** [comment]
Hi @msk4862,
Update your forked repository with the upstream repository :)

**Comment by msk4862** [comment]
@priyansh19 Its showing me `Everything up-to-date`.

**Comment by xandao6** [comment]
Yes, its up-to-date

**Comment by xandao6** [comment]
image
Very Nice!

File: wplay_gui/wplay/src/App.js [hunk 19]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+import "./App.css";
+
+function App() {
+  return (
+    <div className="App">
+      <header className="App-header">
+        <h1>Wplay GUI</h1>
```
Comment (rpotter12) [inline wplay_gui/wplay/src/App.js]: @msk4862 Our project name is whatsapp-play, so please write whatapp-play instead of wplay GUI

Comment (msk4862) [reply wplay_gui/wplay/src/App.js]: @rpotter12 okay

Comment (msk4862) [reply wplay_gui/wplay/src/App.js]: @rpotter12 done.
Please review

**Comment by xandao6** [comment]
@msk4862 I marked the difficulty as medium, as the points will add up to the final PR.

**Pull request merged**